### PR TITLE
Update github action to support linux and macos asset uploading

### PR DESCRIPTION
**Pull request by yorkzero831**
As same as the title

### CODE REVIEW

**Comment by yorkzero831** [comment]
@saharNooby Hi, I found your linux and macos assets are not been uploaded, so I modified your github action

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -59,6 +59,24 @@ jobs:
...
+        if: ${{ ( github.event_name == 'push' && github.ref == 'refs/heads/master' ) || github.event.inputs.create_release == 'true' }}
+        run: |
+          zip rwkv-${{ env.BRANCH_NAME }}-${{ steps.commit.outputs.short }}-bin-linux.zip ./build/librwkv.so
+
+      - name: Upload artifacts
+        if: ${{ ( github.event_name == 'push' && github.ref == 'refs/heads/master' ) || github.event.inputs.create_release == 'true' }}
+        uses: actions/upload-artifact@v3
+        with:
+          path: |
+            rwkv-${{ env.BRANCH_NAME }}-${{ steps.commit.outputs.short }}-bin-linux.zip
```
Comment (saharNooby) [inline .github/workflows/build.yml:78]: I think it needs to specify OS name, `ubuntu` in this case. As I know, there are many different Linuxes, and after building on Ubuntu library may just not work on some.

Is it possible to also include arch and OS version (for Mac OS too)? Maybe there is some env variable for that...

Comment (yorkzero831) [reply .github/workflows/build.yml:78]: Sure, I will handle it asap

Comment (yorkzero831) [reply .github/workflows/build.yml]: @saharNooby Hi, I have updated the file names. 
you could check the file names from https://github.com/yorkzero831/rwkv.cpp/releases/tag/master-a3fe1c6
ubuntu aseet will look like `rwkv-master-a3fe1c6-bin-Linux-Ubuntu-22.04-x86_64.zip`
macos asset will look like `rwkv-master-a3fe1c6-bin-Darwin-macOS-12.6.4-x86_64.zip`
I was thought macos will run on M1 which is arm64, seems github action dosen't support it yet

Comment (saharNooby) [reply .github/workflows/build.yml]: Looks great! One small thing -- Windows zips have `rwkv.dll` in the zip root, without subdirectories. Can Linux/MacOS zips also have no subdirectories for consistency? Or will it break something?

Comment (yorkzero831) [reply .github/workflows/build.yml]: Oh I didn't realize it, will change, give me 10 mins

Comment (yorkzero831) [reply .github/workflows/build.yml]: @saharNooby it shuold be fixed you could verify from https://github.com/yorkzero831/rwkv.cpp/releases/tag/master-5662bf4

**Review by saharNooby — APPROVED** [review]

**Pull request merged**